samples/GstAudioVideoReceiver.c

```diff
@@ -0,0 +1,230 @@
+    if (ret != GST_FLOW_OK) {
+        DLOGE("Error pushing buffer: %s", gst_flow_get_name(ret));
+    }
+    gst_buffer_unref(buffer);
+}
+
+// This function is a callback for the transceiver for every single audio frame it receives
+// It writes these frames to a buffer and pushes it to the `appsrcAudio` element of the
+// GStreamer pipeline created in `receiveGstreamerAudioVideo`. Any logic to modify / discard the frames would go here
+VOID onGstAudioFrameReady(UINT64 customData, PFrame pFrame)
+{
+    GstFlowReturn ret;
+    GstBuffer* buffer;
+    GstElement* appsrcAudio = (GstElement*) customData;
+    if (!appsrcAudio) {
+        DLOGE("Null");
+        return;
+    }
+    if (pFrame == NULL) {
+        DLOGE("Null frame received!, skipping it");
+        return;
+    }
+    buffer = gst_buffer_new_allocate(NULL, pFrame->size, NULL);
+    if (!buffer) {
+        DLOGE("Buffer allocation failed");
+        return;
+    }
+
+    DLOGD("Audio Frame size: %d, presentationTs: %llu", pFrame->size, pFrame->presentationTs);
+    GST_BUFFER_PTS(buffer) = pFrame->presentationTs;
+
+    // Recalculate the byte-rate if not using the default values
+    GST_BUFFER_DURATION(buffer) = gst_util_uint64_scale(pFrame->size, GST_SECOND, DEFAULT_AUDIO_OPUS_BYTE_RATE);
+    if (gst_buffer_fill(buffer, 0, pFrame->frameData, pFrame->size) != pFrame->size) {
+        DLOGE("Buffer fill did not complete correctly");
+        gst_buffer_unref(buffer);
+        return;
+    }
+    g_signal_emit_by_name(appsrcAudio, "push-buffer", buffer, &ret);
+    if (ret != GST_FLOW_OK) {
+        DLOGE("Error pushing buffer: %s", gst_flow_get_name(ret));
+    }
+    gst_buffer_unref(buffer);
+}
+
+// This function is a callback for the streaming session shutdown event. We send an eos to the pipeline to exit the
+// application using this.
+VOID onSampleStreamingSessionShutdown(UINT64 customData, PSampleStreamingSession pSampleStreamingSession)
+{
+    (void) (pSampleStreamingSession);
+    GstElement* pipeline = (GstElement*) customData;
+    gst_element_send_event(pipeline, gst_event_new_eos());
+}
+
+PVOID receiveGstreamerAudioVideo(PVOID args)
+{
+    STATUS retStatus = STATUS_SUCCESS;
+    GstElement *pipeline = NULL, *appsrcAudio = NULL, *appsrcVideo = NULL;
+    GstBus* bus;
+    GstMessage* msg;
+    GError* error = NULL;
+    GstCaps *audiocaps, *videocaps;
+    PSampleStreamingSession pSampleStreamingSession = (PSampleStreamingSession) args;
+    PSampleConfiguration pSampleConfiguration = pSampleStreamingSession->pSampleConfiguration;
+    PCHAR roleType;
+    gchar *videoDescription = "", *audioDescription = "", *audioVideoDescription;
+
+    if (pSampleConfiguration->channelInfo.channelRoleType == SIGNALING_CHANNEL_ROLE_TYPE_MASTER) {
+        roleType = "Master";
+    } else if (pSampleConfiguration->channelInfo.channelRoleType == SIGNALING_CHANNEL_ROLE_TYPE_VIEWER) {
+        roleType = "Viewer";
+    }
+
+    gst_init(NULL, NULL);
```

Comment:
Can we use gst_init_check instead?
https://gstreamer.freedesktop.org/documentation/gstreamer/gst.html?gi-language=c#gst_init_check

Reply:
Done